source/MaterialXTest/MaterialXRuntime/Runtime.cpp

```diff
@@ -1844,12 +1845,64 @@ TEST_CASE("Runtime: commands", "[runtime]")
 
     mx::RtStagePtr stage = api->createStage(MAIN);
 
+    //
+    // Register callbacks for tracking data model changes
+    //
+    auto createPrimCB = [](const mx::RtStagePtr&, const mx::RtPrim&, void* userData)
+    {
+        ++(*reinterpret_cast<size_t*>(userData));
+    };
+    auto removePrimCB = [](const mx::RtStagePtr&, const mx::RtPrim&, void* userData)
+    {
+        ++(*reinterpret_cast<size_t*>(userData));
+    };
+    auto renamePrimCB = [](const mx::RtStagePtr&, const mx::RtPrim&, const mx::RtToken&, void* userData)
+    {
+        ++(*reinterpret_cast<size_t*>(userData));
+    };
+    auto reparentPrimCB = [](const mx::RtStagePtr&, const mx::RtPrim&, const mx::RtPath&, void* userData)
+    {
+        ++(*reinterpret_cast<size_t*>(userData));
+    };
+    auto setAttrCB = [](const mx::RtAttribute&, const mx::RtValue&, void* userData)
+    {
+        ++(*reinterpret_cast<size_t*>(userData));
+    };
+    auto makeConnectionCB = [](const mx::RtOutput&, const mx::RtInput&, void* userData)
+    {
+        ++(*reinterpret_cast<size_t*>(userData));
+    };
+    auto breakConnectionCB = [](const mx::RtOutput&, const mx::RtInput&, void* userData)
+    {
+        ++(*reinterpret_cast<size_t*>(userData));
+    };
```

Comment:
Very simple callback testing, just counting the number of calls we get for each callback, and matching against expected counts below.